### DISCUSSION TITLE
Define request method like POST|GET on element

### DISCUSSION THREAD

**Discussion by ljfreelancer88** [alpinejs/alpine#1351] [4. General]
Hi, is there any way to define request method on the element? Example...
`<a href="#" x-post="/logout" x-data={_csrf: 'csrfGoesHere', uid: 123}>Logout</a>`

Thank you in advance

**Comment by KevinBatdorf** [comment]
What library are you using? What attribute does it want you to put the method in?

**Reply by ljfreelancer88** [reply]
The idea is taken from here https://htmx.org/ but exactly the same.

**Reply by KevinBatdorf** [reply]
Exactly the same as what? Are you trying to define `x-post="/logout"` dynamically so it takes a variable instead of the endpoint? if so, you can bind those the same as any attribute in Alpine, like `x-bind:x-post="endpoint"`. where you have `endpoint` defined in your `x-data`
